src/endpoint/s3/ops/s3_get_bucket_notification.js

```diff
@@ -12,34 +12,21 @@ async function get_bucket_notification(req) {
         bucket_name: req.params.bucket,
     });
 
-    result = _.cloneDeep(result);
-
-    const TopicConfiguration = [];
-
-    //adapt to aws cli structure
-    if (result && result.length > 0) {
-        for (const conf of result) {
-            conf.Event = conf.event;
-            conf.Topic = conf.topic;
-            conf.Id = conf.id;
-            delete conf.event;
-            delete conf.topic;
-            delete conf.id;
-
-            TopicConfiguration.push({TopicConfiguration: conf});
-        }
-    }
-
-    const reply = result && result.length > 0 ?
-        {
-            //return result inside TopicConfiguration tag
-            NotificationConfiguration:
-                TopicConfiguration
-        } :
-        //if there's no notification, return empty NotificationConfiguration tag
-        { NotificationConfiguration: {} };
-
-    return reply;
+    //adapt to structure that xml_utils.encode_xml() will then
+    //encode into aws compliant reply
+    return {
+        NotificationConfiguration: [_.map(result, t => ({
```

Comment:
no need to wrap the map with [] because it always returns an array
```suggestion
        NotificationConfiguration: _.map(result, t => ({
```